theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -63,3 +108,310 @@ def repoclosures(repo, releases, version) {
+            repos["${suite}-${component}"].deb_paths.add("${deb_path}/*deb}")
+        }
+    }
+
+    repos.each { key, value ->
+        if (value.type == 'stage') {
+            steps.rsync_packages[key] = {
+                rsync_to_debian_stage(value.suite, value.component, value.deb_paths)
+            }
+        } else {
+            steps.rsync_packages[key] = {
+                rsync_to_debian_release(value.suite, value.component, value.deb_paths)
+            }
+        }
+    }
+
+    return steps
+}
+
+def deb_build_dir(type, project, os = null) {
+    def build_dir
+
+    if (type == 'core') {
+        build_dir = "debian/${os}/build-${project}"
+    } else if (type == 'dependency') {
+        build_dir = "dependencies/${os}/build-${project}"
+    } else if (type == 'plugin') {
+        build_dir = "plugins/build-${project}"
+    } else {
+        echo "Unsupported type specified: ${type}"
+        exit 1
+    }
+
+    return build_dir
+}
+
+def setup_sources_deb(type, project, os, version, repoowner, pull_request) {
+
+    if (type == 'core') {
+        setup_sources_core(project, os, version, repoowner, pull_request)
+    } else if (type == 'dependency') {
+        setup_sources_dependency(project, os, version, repoowner, pull_request)
+    } else if (type == 'plugin') {
+        setup_sources_plugin(project, os, version, repoowner, pull_request)
+    } else {
+        echo "Unsupported type specified: ${type}"
+        exit 1
+    }
+
+}
+
+def setup_sources_core(os, project, version, repoowner, pull_request = false) {
+    def package_version = debian_package_version("debian/${os}/${project}/changelog")
+    def build_dir = deb_build_dir('core', project, os)
+
+    package_version = package_version.replaceAll('~rc', '-rc')
+    last_commit = package_version
+
+    dir(build_dir) {
+        if (version =~ 'develop') {
+            def url_base = 'https://ci.theforeman.org'
+            def job_name = nightly_jenkins_job
+            def job_id = nightly_jenkins_job_id
+            def job_url = "${url_base}/job/${job_name}/${job_id}"
+
+            sh """
+              wget "${job_url}/artifact/*zip*/archive.zip"
+              unzip archive.zip
+              mv archive/pkg/*bz2 ${project}_${package_version}.orig.tar.bz2
+
+              # Set this in case we need it
+            """
+
+            last_commit = sh(returnStdout: true, script: "curl \"${job_url}/api/json\" | jq -r '.actions[].lastBuiltRevision.SHA1 | values'").trim()
+        } else {
+            sh """
+              # Download sources
+              wget https://downloads.theforeman.org/${project}/${project}-${package_version}.tar.bz2 https://downloads.theforeman.org/${project}/${project}-${package_version}.tar.bz2.sig
+              mv ${project}-${package_version}.tar.bz2 ${project}_${package_version}.orig.tar.bz2
+            """
+        }
+
+        sh """
+            # Unpack
+            tar xvjf ${project}_${package_version}.orig.tar.bz2
```

Comment:
```suggestion
            tar xf ${project}_${package_version}.orig.tar.bz2
```

* `j` is not needed as `tar` can figure out the compression on its own
* `v` makes the build log so noisy, that I'd prefer if we would drop it